setuptools/tests/test_easy_install.py

```diff
@@ -466,7 +466,7 @@ def test_setup_requires_honors_fetch_params(self, mock_index, monkeypatch):
                     with contexts.environment(PYTHONPATH=temp_install_dir):
                         cmd = [
                             sys.executable,
-                            '-m', 'setup',
```

Comment:
This test seems to rely on the fact that it runs from the project root (that contains a `setup.py`) file.

Reply:
Wow. That test was really janky, relying on Setuptools' own setup.py (with all its distribution setup) just to ignore it and invoke easy_install. Yes, I believe this change is better.

Reply:
Moreover: `runpy` also injects the CWD into `PYTHONPATH` (which will effectively be prepended to `temp_install_dir` that the CM above puts there).

Reply:
I am afraid the only way of preventing this from happening would be either adopting a src-layout or using the `changedir` trick proposed in the original issue.

The problem with the `changedir` being that it breaks `pytest-enabler` and (a single) doctest...